lib/providers/token.js

```diff
@@ -176,10 +176,26 @@ class TokenProvider {
         };
 
         return this.data;
+      }).catch((err) => {
+        // Only consider StatusCodeError to be a failing condition. Anything else could be network
+        // related or something else that we want to retry. StatusCodeError comes from request-promise which Vaulted
+        // uses to hit Vault.
+        if (err.name === 'StatusCodeError') {
+          throw err;
```

Comment:
Let's get this PR merged, then I'll add a bunch of logging elsewhere and open a new one.